TSServer does not print logs when custom node is used

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- 🕮 Read our guide about submitting issues: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions -->
<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Insiders build to see if your issue has already been fixed: https://code.visualstudio.com/insiders/ -->
<!-- 💡 Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information. -->
<!-- 🔧 Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: Yes

<!-- 🪓 If you answered No above, use 'Help: Start Extension Bisect' from Command Palette to try to identify the cause. -->
<!-- 📣 Issues caused by an extension need to be reported directly to the extension publisher. The 'Help > Report Issue' dialog can assist with this. -->
- VS Code Version: 1.86.0
- OS Version: Darwin x64 23.2.0

Steps to Reproduce:

1. enable custom node:
```json
  "typescript.tsserver.log": "verbose",
  "typescript.tsserver.enableTracing": true,
  "typescript.tsserver.nodePath": "/Users/xxx/.nvm/versions/node/v19.9.0/bin/node",
```
2. open any ts project. `code --user-data-dir ~/.vscode-debug/   --disable-extensions  my_project`. The log file is not generated actually.

```
2024-02-08 11:13:31.997 [info] Killing TS Server
2024-02-08 11:13:31.998 [info] Starting TS Server
2024-02-08 11:13:31.998 [info] Using tsserver from: /Users/xxx/.vscode/extensions/ms-vscode.vscode-typescript-next-5.4.20240206/node_modules/typescript/lib/tsserver.js
2024-02-08 11:13:31.998 [info] Using Node installation from /Users/xxx/.nvm/versions/node/v19.9.0/bin/node to run TS Server
2024-02-08 11:13:31.998 [info] <syntax> Log file: /Users/xxx/Library/Application Support/Code/logs/20240207T135733/window4/exthost/vscode.typescript-language-features/tsserver-log-2o1bl4/tsserver.log
2024-02-08 11:13:31.998 [info] <syntax> Forking...
2024-02-08 11:13:31.998 [info] <syntax> Starting...
2024-02-08 11:13:31.999 [info] <semantic> Log file: /Users/xxx/Library/Application Support/Code/logs/20240207T135733/window4/exthost/vscode.typescript-language-features/tsserver-log-gG0wun/tsserver.log
2024-02-08 11:13:31.999 [info] <semantic> Forking...
2024-02-08 11:13:31.999 [info] <semantic> Starting...
2024-02-08 11:13:32.181 [error] TSServer exited. Code: null. Signal: SIGTERM
```

After attaching to the tsserver with `TSS_DEBUG_BRK=5667`, I see a strange error:

```
Uncaught Error Error: Cannot find module '/Applications/Visual'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1090:15)
    at Module._load (node:internal/modules/cjs/loader:934:27)
    at executeUserEntryPoint (node:internal/modules/run_main:83:12)
    at <anonymous> (node:internal/main/run_main_module:23:47)
```

> After attaching to the tsserver with `TSS_DEBUG_BRK=5667`, I see a strange error:
> 
> ```
> Uncaught Error Error: Cannot find module '/Applications/Visual'
>     at Module._resolveFilename (node:internal/modules/cjs/loader:1090:15)
>     at Module._load (node:internal/modules/cjs/loader:934:27)
>     at executeUserEntryPoint (node:internal/modules/run_main:83:12)
>     at <anonymous> (node:internal/main/run_main_module:23:47)
> ```

I set "typescript.tsserver.nodePath" to be a script which echos its CLI args and it also shows that the spaces aren't be escaped properly in `/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js`

This looks like a bug in https://github.com/microsoft/vscode/pull/191019 (@gabritto) which will likely be broken for any Mac user.

--- a/extensions/typescript-language-features/src/tsServer/serverProcess.electron.ts
+++ b/extensions/typescript-language-features/src/tsServer/serverProcess.electron.ts
@@ -278,8 +278,7 @@ export class ElectronServiceProcessFactory implements TsServerProcessFactory {
 		}
 
 		const childProcess = execPath ?
-			child_process.spawn(JSON.stringify(execPath), [...execArgv, tsServerPath, ...runtimeArgs], {
-				shell: true,
+			child_process.spawn(execPath, [...execArgv, tsServerPath, ...runtimeArgs], {
 				windowsHide: true,
 				cwd: undefined,
 				env,

--- a/extensions/typescript-language-features/src/tsServer/spawner.ts
+++ b/extensions/typescript-language-features/src/tsServer/spawner.ts
@@ -234,7 +234,7 @@ export class TypeScriptServerSpawner {
 					tsServerLog = { type: 'file', uri: logFilePath };
 
 					args.push('--logVerbosity', TsServerLogLevel.toString(configuration.tsServerLogLevel));
-					args.push('--logFile', `"${logFilePath.fsPath}"`);
+					args.push('--logFile', logFilePath.fsPath);
 				}
 			}
 		}

